sql/catalyst/src/main/java/org/apache/spark/sql/connector/expressions/GeneralScalarExpression.java

```diff
@@ -20,22 +20,42 @@
 import java.io.Serializable;
 
 import org.apache.spark.annotation.Evolving;
+import org.apache.spark.sql.connector.util.V2ExpressionSQLBuilder;
 
 /**
- * The general SQL string corresponding to expression.
+ * The general representation of SQL scalar expressions, which contains the upper-cased
+ * expression name and all the children expressions.
  *
  * @since 3.3.0
  */
 @Evolving
-public class GeneralSQLExpression implements Expression, Serializable {
-    private String sql;
+public class GeneralScalarExpression implements Expression, Serializable {
+  private String name;
+  private Expression[] children;
+  private String sql;
 
-    public GeneralSQLExpression(String sql) {
-        this.sql = sql;
-    }
+  public GeneralScalarExpression(String name, Expression[] children) {
+    this.name = name;
+    this.children = children;
+  }
 
-    public String sql() { return sql; }
+  public String name() { return name; }
+  public Expression[] children() { return children; }
 
-    @Override
-    public String toString() { return sql; }
+  @Override
+  public String toString() {
+    if (sql == null) {
+      V2ExpressionSQLBuilder builder = new V2ExpressionSQLBuilder();
+      try {
+        this.sql = builder.build(this);
+      } catch (Throwable e) {
+        // Attempt to get SQL failed.
+      }
+    }
+    if (sql == null) {
+      return name + "(" + children.toString() + ")";
```

Comment:
`Expression[].toString` returns a meaningless string, I think it's better to use `children.mkString(", ")` (please use proper java 8 lambda syntax)